app/src/lib/file-system.ts

```diff
@@ -0,0 +1,20 @@
+import * as fileSystem from 'fs-extra'
+import ErrnoException = NodeJS.ErrnoException
+
+interface IErrorCallbackType { (err: ErrnoException): void}
+
+
+/** this function wraps the filesystem mkdir function and silently returns if
+ * the directory already exists. If a different error was thrown, it's passed to an error handling callback
+ *
+ * @param directoryPath the path of the directory the user wants to create
+ * @param errorCallback function to be called in the event of an unexpected error
+ */
+export async function mkdirIfNeeded (directoryPath: string, errorCallback: IErrorCallbackType) {
```

Comment:
No real need to break `errorCallback` out to it's own type. This seems readable enough:

```ts
export async function mkdirIfNeeded(directoryPath: string, errorCallback: (err: ErrnoException) => void) {
```